modules/hal_nxp/fsl_os_abstraction_zephyr.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "fsl_os_abstraction.h"
+
+#ifdef CONFIG_NXP_FSL_OSA
+
+#ifdef CONFIG_NXP_FSL_OSA_HEAP
+/* calculate the required heap size of mcux usb host controller */
+#ifdef CONFIG_USB_UHC_NXP_MCUX
```

Comment:
OSA should not have IP specific code  or configuration.
Please remove the USB code from the Common OSA, and find other solution.

Reply:
Thanks Andrej, My thought is: The osa can define the heap size based on all the enabled modules that uses the osa heap.  Do you have any suggestions?

Reply:
It should be a project configuration parameter.

Reply:
I think it is better to put the similar logic to Kconfig as follow.
```
config NXP_FSL_OSA_HEAP_SIZE
	int
	default 2500 if CONFIG_USB_UHC_NXP_MCUX && CONFIG_NXP_OTHER_MODULE
	default 1500 if CONFIG_USB_UHC_NXP_MCUX
	default 1000 if CONFIG_NXP_OTHER_MODULE
	default 0
```
since there is no CONFIG_NXP_OTHER_MODULE now. the code is:
```
config NXP_FSL_OSA_HEAP_SIZE
	int
	default 1500 if CONFIG_USB_UHC_NXP_MCUX
	default 0
```

What do you think?

Reply:
Can we use already existing Zephyr heap definitions?

Reply:
Do you mean the CONFIG_HEAP_MEM_POOL_ADD_SIZE_ ${DOTCONFIG}? Different modules can increase the heap size by module's self macro. If we don't need the osa heap to be non-cachable (usb controller driver doesn't need it now) I think we can do it. In current USB stack, the system heap (k_malloc) is not enabled default, so I need to enable it if nxp usb controller is enabled. I can change the PR for review if this is your suggestion.

Reply:
Thanks Andrej